pkg/firewall/client.go

```diff
@@ -2,66 +2,124 @@ package firewall
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/giantswarm/microerror"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
 	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type Client struct {
-	fwService *compute.FirewallPoliciesClient
+	fwService *compute.FirewallsClient
 	k8sClient client.Client
 }
 
-func NewClient(fwService *compute.FirewallPoliciesClient, k8sClient client.Client) *Client {
+func NewClient(fwService *compute.FirewallsClient, k8sClient client.Client) *Client {
 	return &Client{
 		fwService: fwService,
 		k8sClient: k8sClient,
 	}
 }
 
+const (
+	ProtocolTCP = "tcp"
+	PortSSH     = "22"
+)
+
 func (c *Client) CreateBastionFirewallRule(ctx context.Context, cluster *capg.GCPCluster) error {
-	// TODO
+	fwName := bastionFirewallPolicyRuleName(cluster.Name)
```

Comment:
I'd add some logging to this function, otherwise it's hard to know what the operator is doing.